dotnet/api/Presentation/Web/Controllers/Api/V1/UserLogins/UserLoginsController.cs

```diff
@@ -0,0 +1,183 @@
+            _cookieAuthenticationConfiguration = cookieAuthenticationConfiguration;
+            _httpContext = httpContextAccessor.HttpContext;
+            _logger = logger;
+            _mapper = mapper;
+            _userLoginConductor = userLoginConductor;
+            _userReadConductor = userReadConductor;
+            _userRoleReadConductor = userRoleReadConductor;
+        }
+
+        #endregion Constructors
+
+        #region POST
+
+        [AllowAnonymous]
+        [HttpPost]
+        public async Task<IActionResult> Create(
+            [FromBody] UserLoginDto dto,
+            [FromQuery] long? roleId = null
+        )
+        {
+            var userName = dto.UserName?.ToLower();
+            var authenticationResult = _userLoginConductor.Authenticate(userName, dto.Password);
+            if (authenticationResult.HasErrorsOrResultIsNull())
+            {
+                return HandleFailedLogin(userName, authenticationResult.Errors);
+            }
+
+            var user = authenticationResult.ResultObject;
+            var createResult = CreateUserLogin(userName, user.Id, roleId);
+            if (createResult.HasErrorsOrResultIsNull())
+            {
+                return InternalError<UserLoginDto>(createResult.Errors, _logger);
+            }
+
+            var userLogin = createResult.ResultObject;
+
+            await SignInUser(userLogin);
+
+            return Created(userLogin.Id, _mapper.Map<UserLoginDto>(userLogin));
+        }
+
+        #endregion POST
+
+        #region Private Methods
+
+        private IResult<UserLogin> CreateUserLogin(
+            string userName,
+            long? userId = null,
+            long? roleId = null,
+            bool isSuccessful = true
+        )
+        {
+            var userLogin = new UserLogin
+            {
+                FailedAttemptCount = isSuccessful ? 0 : 1,
+                Ip = IpAddress,
+                IsSuccessful = isSuccessful,
+                RoleId = roleId,
+                ServerName = Environment.MachineName,
+                UserAgent = UserAgent,
+                UserId = userId,
+                UserName = userName
+            };
+
+            return _conductor.Create(userLogin, userId);
+        }
+
+        private IActionResult HandleFailedLogin(string userName, List<IError> errors)
+        {
+            // Find the last UserLogin for this Username
+            var userLoginResult = _conductor.FindAll(e => e.Ip == IpAddress && e.UserName == userName);
+            var userLogin = userLoginResult.ResultObject?.OrderByDescending(e => e.CreatedOn)?.FirstOrDefault();
+
+            // If last UserLogin is null or was successful create a new failed one.
+            if (userLogin == null || userLogin.IsSuccessful)
+            {
+                var createResult = CreateUserLogin(userName, isSuccessful: false);
+                if (createResult.HasErrorsOrResultIsNull())
+                {
+                    errors = createResult.Errors;
```

Comment:
Should this be returning an `InternalError` if it fails to create the UserLogin?

Reply:
Yeah, that's a good point! Updating it